practices/unit_tests.md

```diff
@@ -0,0 +1,93 @@
+
+Unit testing can be done manually but it is usually automated. In terms of automated tests, there is a concept called **the test pyramid** that shows how to efficiently balance the automated tests.
+
+![Test Pyramid](/files/unit_tests.png)
+
+[Amazon Alexa: Unit Testing: Creating Functional Alexa Skills](https://developer.amazon.com/it/blogs/alexa/post/35bdad3d-57c8-4623-88c6-815540697af5/unit-testing-create-functional-alexa-skills)
+
+- **Unit Test**
+		Unit tests typically test functions, methods, and classes. They are short, cheap and fast to write and maintain. They are focused on one thing/function. They are not dependent on any external systems. When a unit test fails, it is easy to identify the problem.
+
+- **Integration Test**
+		Integration test focuses on the point where individual units are combined and tested as a group. The purpose is to test if the interaction between units works.
+
+- **End to End Test**
+		End to end tests cover the whole application - from start to finish. They are more expensive and slower to run.
+
+The pyramid shows you that the more high-level you get, the fewer tests you should perform. Write a lot of small and fast unit tests, some integration tests and just a few high-level tests. To maintain this balance, use [continuous delivery](/issues/continuous_delivery) - a practice where you automatically ensure your product is ready to be released into production.
+
+## Why You Might Want the Unit Tests
+
+Unit testing helps you with:
+- maintaining your code
+- catching defects introduced due to the code change
+- lower the potentially harmful impact of changes to your code
+- reusing the code
+- faster development
+- lowering the cost of fixing defects on lower testing level
+- a code documentation because the unit tests describe what your product does
+
+## Issues the Unit Tests Solves
+
+- [Poor code quality](/issues/poor-code-quality)
+- [Increased cost](/issues/increased-cost)
+- Meaningless work
+- Insufficient performance
+- [demotivated team](/issues/demotivated-team)
+
+## How to Implement the Unit Tests
+
+- Find a tool or a framework suitable for your programming language such as [NCrunch](https://www.ncrunch.net/), [EmbUnit](https://www.embunit.com/), or [HtmlUnit](http://htmlunit.sourceforge.net/).
+- Do not try to reach 100% unit test coverage. Focus on the behaviors of the tested system.
+-  The unit test name should explain what was tested.
+- Perform unit tests continuously and frequently.
+
+**Unit Testing Best Practice:**
+- **Arrange:**  Set up the test. Make mocking tests for isolationg the behavior.  Use inversion of control.
```

Comment:
See the preview: https://preview.developerexperience.io/practices/unit-tests the list is not formatted correctly because a blank line before the list is missing.